modules/data.remote/R/remote_process.R

```diff
@@ -0,0 +1,1054 @@
+    scale <- as.double(settings$remotedata$scale)
+    scale <- format(scale, nsmall = 1)
+  }
+  projection       <- settings$remotedata$projection
+  qc               <- settings$remotedata$qc
+  if (!is.null(qc)) {
+    qc <- as.double(settings$remotedata$qc)
+    qc <- format(qc, nsmall = 1)
+  }
+  algorithm        <- settings$remotedata$algorithm
+  credfile         <- settings$remotedata$credfile
+  pro_mimetype     <- settings$remotedata$pro_mimetype
+  pro_formatname   <- settings$remotedata$pro_formatname
+  out_get_data     <- settings$remotedata$out_get_data
+  out_process_data <- settings$remotedata$out_process_data
+  overwrite        <- settings$remotedata$overwrite
+  if (is.null(overwrite)) {
+    overwrite <- FALSE
+  }
+  
+  
+  PEcAn.logger::severeifnot("Check if siteid is of numeric type and is not NULL",
+                            is.numeric(siteid))
+  PEcAn.logger::severeifnot("Check if outdir is of character type and is not NULL",
+                            is.character(outdir))
+  PEcAn.logger::severeifnot("Check if source is of character type and is not NULL",
+                            is.character(source))
+  # PEcAn.logger::severeifnot("Source should be one of gee, appeears", source == "gee" || source == "appeears")
+  # collection validation to be implemented
+  if (!is.null(projection)) {
+    PEcAn.logger::severeifnot("projection should be of character type",
+                              is.character(projection))
+  }
+  if (!is.null(algorithm)) {
+    PEcAn.logger::severeifnot("algorithm should be of character type",
+                              is.character(algorithm))
+  }
+  if (!is.null(credfile)) {
+    PEcAn.logger::severeifnot("credfile should be of character type",
+                              is.character(credfile))
+  }
+  PEcAn.logger::severeifnot(
+    "Check if out_get_data is of character type and is not NULL",
+    is.character(out_get_data)
+  )
+  if (!is.null(out_process_data)) {
+    PEcAn.logger::severeifnot("out_process_data should be of character type",
+                              is.character(out_process_data))
+  }
+  
+  
+  dbcon <- PEcAn.DB::db.open(settings$database$bety)
+  on.exit(PEcAn.DB::db.close(dbcon), add = TRUE)
+  
+  # collection dataframe used to map Google Earth Engine collection names to their PEcAn specific names
+  collection_lut <- data.frame(
+    stringsAsFactors = FALSE,
+    original_name = c(
+      "LANDSAT/LC08/C01/T1_SR",
+      "COPERNICUS/S2_SR",
+      "NASA_USDA/HSL/SMAP_soil_moisture"
+    ),
+    pecan_code = c("l8", "s2", "smap")
+  )
+  getpecancode <- collection_lut$pecan_code
+  names(getpecancode) <- collection_lut$original_name
+  
+  if (source == "gee") {
+    collection = unname(getpecancode[collection])
+  }
+  
+  # construct raw file name
+  raw_file_name <-
+    construct_raw_filename(collection, siteid_short, scale, projection, qc)
+  
+  
+  # check if any data is already present in the inputs table
+  dbstatus <-
```

Comment:
could you return a named list from this function so that you can access elements by their names? using indices is harder to maintain and more error-prone

Reply:
sorry should have thought about this, will change.